civis/io/_tables.py

```diff
@@ -622,17 +622,16 @@ def dataframe_to_civis(df, database, table, api_key=None, client=None,
     if archive:
         warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
                       "Use `hidden` instead.", FutureWarning)
-    buf = six.BytesIO()
-    if six.PY3:
-        txt = io.TextIOWrapper(buf, encoding='utf-8')
-    else:
-        txt = buf
-    df.to_csv(txt, encoding='utf-8', index=False, **kwargs)
-    txt.flush()
-    buf.seek(0)
+
+    with TemporaryDirectory() as tmp_dir:
+        tmp_path = os.path.join(tmp_dir, 'dataframe_to_civis.csv')
+        with open(tmp_path, 'w') as fin:
+            df.to_csv(fin, encoding='utf-8', index=False, **kwargs)
```

Comment:
Is there a reason why you open the file yourself rather than giving the filename to `pandas`?

Reply:
No; good catch